src/main/java/io/github/thebusybiscuit/slimefun4/core/attributes/DamageableItem.java

```diff
@@ -69,4 +81,26 @@ default void damageItem(@Nonnull Player p, @Nullable ItemStack item) {
         }
     }
 
+    /**
+     * This method will randomly decide if the item should be damaged or not
+     * This does not damage the item, it is called by {@link #damageItem(Player, ItemStack)} to randomly generate a boolean
+     *
+     * @param unbreakingLevel
+     *                                 The {@link Integer} level of the unbreaking enchantment
+     * @throws IllegalArgumentException
+     *                                 Thrown when {@link #itemType()} doesn't return a valid {@link UnbreakingAlgorithms} value
+     * @return Whether you should keep the item undamaged
+     *
+     */
+    default boolean evaluateUnbreakingEnchantment(int unbreakingLevel) {
+        switch (itemType()){
+            case TOOLS:
+                return !(Math.random() < (1.0 / (unbreakingLevel + 1)));
+            case ARMORS:
+                return !(Math.random() < 0.6 + (0.4 / (unbreakingLevel + 1)));
```

Comment:
I think that these formulas should rather be stored inside the respective enum constant instead of relying on a switch/case here.

Reply:
THen we could simplify:

```java
default boolean evaluateUnbreakingEnchantment(int unbreakingLevel) {
    return UnbreakingAlgorithm.TOOLS.evaluate(unbreakingLevel);
}
```